docs/DeveloperGuide.md

```diff
@@ -757,4 +757,128 @@ Manage finances containing multiple payment methods faster than a typical mouse/
+3. Adding income source
+   1. Test case: `add /i /a 200 /c Allowance`
+      Expected: Income source added to account. Message displaying summary of new
+      income source is displayed. 
+   2. Test case: Other valid category (Salary, Investment, Others)
+      Expected: Income source added to account. Message displaying summary of new
+      income source is displayed.
+   3. Test case: Invalid category 
+      Expected: Add command is rejected, with an appropriate error message.
+   4. Test case: Invalid amount (not a positive integer)
+      Expected: Add command is rejected, with an appropriate error message.
+4. Adding expenditure paid for with credit card
+   1. Prerequisite: A credit card named `DBS` has been added, with an appropriately high
+      credit limit.
+   2. Test case: `add /e /pm DBS /c Personal /d Gift to friend /a 10.00 /t 01/01/2022`
+      Expected: Expenditure added to account. Message displaying summary of new
+      expenditure, along with updated balance of credit card, is displayed.
+   3. Test case: Non-existent credit card
+      Expected: Add command is rejected, with an appropriate error message.
+   4. Test case: Amount exceeding credit card's limit
+      Expected: Add command is rejected, with an appropriate error message.
+
+### Updating and deleting data
+
+1. Updating and deleting expenditures
+   1. Prerequisite: There is at least 1 expenditure.
+   2. Test case: `update /e 1 /pm Cash /c Entertainment /d PS5 /a 899.00 /t 25/03/2022`
+      Expected: First expenditure is updated accordingly, with a message summarizing
+      the new data.
+2. Updating and deleting credit cards
+   1. Prerequisite: There are at least 4 credit cards.
+   2. Test case: `update /cc 4 /n OCBC /cb 2.00 /cl 30000`
+      Expected: Fourth credit card is updated accordingly, with a message summarizing
+       the new data.
+3. Updating and deleting income sources
+   1. Prerequisite: There are at least 2 income sources.
+   2. Test case: `update /i 2 /a 500 /c Investment`
+      Expected: Second income source is updated accordingly, with a message summarizing
+       the new data.
+4. Update edge cases
+   1. Test case: Invalid index into list (not integer, negative, out of bounds, etc)
+      Expected: Update command is rejected, with an appropriate message.
+   2. Note that all input format edge cases from `add` also apply here.
+
+### Displaying summary information
+
+1. Listing information
+   1. Prerequisite for ii, iii, iv: Some expenditures, credit cards, and income sources have already been `add`ed.
+   2. Test case: `list /cc`
+      Expected: A list of credit cards is shown. For each credit card, its name,
+      cashback percentage, cashback gained, card limit, and current remaining balance
+      is displayed.
+   3. Test case: `list /i`
+      Expected: A list of income sources is shown. For each income source, its
+      amount and category is displayed.
+   4. Test case: `list /e`
```

Comment:
Perhaps can add the optional {DATE} field. if the user would like to search by date.